### PR TITLE
Engineering and Vending Improvements

### PR DESCRIPTION
**Pull request by Arctrooper8**
## About The Pull Request

Currently engineers spawn with a general circuit which requires a multitool to change into something useful, considering the only circuits engineers will actually use are APC circuits I have replaced their starting general circuit with an APC circuit. In fact I have completely deleted the general circuit. Engi-vend (electronics vendor) now dispenses as many APC boards and Multitools as you need (10 cells too, but those were already there for those who knew where to look), and because of this multitools and APC boards cannot be purchased in the Squad Engineer point vendor anymore. 

The Engi-Vend and TG Engineer System vendor (Sentry vendor) have been moved to public places on all ships for better standardization and ease of access.
(Pillar of Spring. You can also see I placed the button on the wall because it looks nicer, alongside moving the windows since the vendors now block the view)
AAAAAAAAA

(Minerva)

(Sulaco)
New Sulaco

(Thesus, it might've been like this from the start however the sulaco's public TG Engineer System vendor was the inspiration for all the moving of vendors)
Thesus Engi

(Twin Pillars, yes I even did the awful HvH ship.)
Twin Pillars

And with the vendors public and easily accessible alongside noticeable, I used the new space in engi prep.
Pillars new engi prep

The metal construction menu now looks much nicer  with old unused recipes being removed
New metal menu

Finally, contraband is dead. Most of it has been moved into the vendor without needing to be hacked.
Training Toxins

Oh I made all the vending machine lists look nice too since I killed contraband and had to move multi and circuit into the engi vend, bonus refactor Yay! (or is that just code change?)

Super ultra marine power creep secret commit:
Office chair
(The office chair has it's own description now, I'm sure people will appreciate this when using them for "Office work")

## Why It's Good For The Game

QOL is always good for the game when it makes people's lives easier. When engineers only ever build one thing it makes sense to give them that exact circuitboard instead of a board they might not have the tool for if they brought along a tool pouch instead of belt (Looking at you PC engis).

The metal menu has been desperately in need of an update since it has not changed since _Dinosaurs ruled the earth!_ having strange recipes such as teal comfy chair (RGB coloured comfy chair that looks awful) and metal baseball bat (You can still make wood bats if you really want to RP as some true xeno slugger). Also by having the important structures such as cades and razorwire at the top new and veteran Engis can more easily find what they want.

Bravemole says he doesn't like contraband and being forced to fumble hack through each wire just for something you want is not fun.

Moving the vendors into public areas makes them more easily seen and used by all marines (no more forgetting sentries!). And removing the tool vendor from the Engineer prep room allows for more useful vendors when engineers already have a tool belt and tool pouches that come fully loaded.

The vending refactor change makes all the lists easier to read and makes me happy when I look at them.

Oh and I killed the old Tool storage vendor because it had no niche at all and mostly sold stuff that is no longer useful such as manipulators, JPR told me their old war stories about actually having to build autolathes then told me to take the old vendor out back and mercy kill it. (It's also only used like five times throughout all the maps; believe me, I checked.)

Overall this PR may seem like a lot but it's mostly a bunch of small QOL changes revolving around vendors and circuits. Small QOL can often be overlooked in favour of new features or big balance changes. Yet it is no less important as even slightly reducing tedium can vastly improve the user experience.

## Changelog
:cl:
balance: Contrand has now been removed from all vendors, their contents are now (mostly) available without being hacked.
del: Deleted Tool storage vendor at JPR's request, and the general circuit (not at JPR's request for that one though)
qol: Engineers now start with an APC circuit and have had the iron build menu improved, office chairs now have a description too!
refactor: Makes vending lists nicer to look at while removing contraband lists
/:cl:

### CODE REVIEW

**Comment by Arctrooper8** [comment]
Hooray! All check have passed! @BraveMole I finished my PR that we discussed on discord. And I also forgot how to request review, my apologies.

**Comment by Arctrooper8** [comment]
Readded rods since you can cross lava with them as pointed out to me by Pike Allen and Pomothgranate.
Rods return!

**Comment by Mrrpip** [comment]
It'd be a shame to see light fixtures go, I quite liked fitting a light and then using a light replacer on it.

**Comment by Arctrooper8** [comment]
Alright fair, I had an idea to make lights and light replacers more useful for a later PR anyway!

**Comment by Moggram** [comment]
I don't mind all the vending and moving stuff round in the metal menu, but remove recipes (IMO) is repulsing. You should be adding more, not taking away from what engineers can do. 
Also the general circuit board is nice to have because what if I do want to make a door or something. Yes I know atmo is dead on TGMC and we don't need air or fire alarms, but one day we may want to try Monsteratmos.
When I play as combat engineer, I want to feel like a engineer. Maybe I am the only one that feels this way.

**Comment by Moggram** [comment]
>
> If you want to add fluff things like that, that's perfectly fine. The issue is that it's in the middle of usefull things, the worst part is when you have to look for the fucking cade among light fightures and other funny but useless stuff.

This was in the first part of my post. I don't mind moving stuff round in the metal menu. I get that part, but is it not the ship eng job to fix the stuff on the ship that the marines break into? 

> 
> You'll never ever make a door, and maybe we will see one engineer doing a door among 1000 games. Let's just remove it

Airlocks with glass windows would more useful if lasguns could shoot through them.

**Comment by Notamaniac** [comment]
Maybe you could implement a sub menu with all the removed recipes, that way you both remove clutter and keep the ability to build machines and stuff.

**Comment by RipGrayson** [comment]
> I don't mind all the vending and moving stuff round in the metal menu, but remove recipes (IMO) is repulsing. You should be adding more, not taking away from what engineers can do. 

While I (mostly) agree with this statement, I have to point out that each item we leave has a certain (however small) maintainer cost to keeping it working/free of bad code/exploit free etc. The question becomes, what's the point of doing that work for something that isn't going to be used except 1 in 1000 rounds?

Ideally somebody would go through the removed items and come up with a better purpose for them, but barring that do we have a reason to keep them in their current state?

**Comment by Losenis** [comment]
Do not remove the girder construction recipe, because that's an actual defensive tool that trades huge metal costs in exchange for making it only fully dealt with by a combination of acid caste + big xeno to destroy the wall. If you really want to make the menu neater just make it a collapsible/separate menu labelled "misc" or something, but players shouldn't be fucked out of additional options for those freak cases just because of menu aesthetics that many don't even really care about since they'll probably have the cade button location memorized anyway.

File: code/game/objects/machinery/vending/vending_types.dm [hunk 1]
```diff
@@ -381,32 +461,21 @@
...
-	desc = "A large storage machine containing various tools and devices for general repair."
-	icon_state = "engi"
-	icon_deny = "engi-deny"
-	req_access = list(ACCESS_MARINE_ENGINEERING)
-	products = list(/obj/item/clothing/head/hardhat = 4,/obj/item/storage/belt/utility = 4,/obj/item/clothing/glasses/meson = 4,/obj/item/clothing/gloves/yellow = 4,
-					/obj/item/tool/screwdriver = 12,/obj/item/tool/crowbar = 12,/obj/item/tool/wirecutters = 12,/obj/item/multitool = 12,/obj/item/tool/wrench = 12,
-					/obj/item/t_scanner = 12, /obj/item/cell = 8, /obj/item/tool/weldingtool = 8,/obj/item/clothing/head/welding = 8,
-					/obj/item/light_bulb/tube = 10, /obj/item/stock_parts/scanning_module = 5,/obj/item/stock_parts/micro_laser = 5,
-					/obj/item/stock_parts/matter_bin = 5,/obj/item/stock_parts/manipulator = 5,/obj/item/stock_parts/console_screen = 5)
+	products = list(
```
Comment (BraveMole) [inline code/game/objects/machinery/vending/vending_types.dm:472]: mesons are still used, give belts as well. I don't remember there being belts in other vendors

File: code/game/objects/machinery/vending/marine_vending.dm [hunk 2]
```diff
@@ -963,10 +953,9 @@
 		/obj/structure/closet/crate/mortar_ammo/mortar_kit = 1,
 		/obj/item/storage/box/sentry = 3,
 		/obj/item/storage/box/tl102 = 1,
+		/obj/item/cell/super = 1,
```
Comment (BraveMole) [inline code/game/objects/machinery/vending/marine_vending.dm:956]: yeah just remove this one

File: code/game/objects/machinery/vending/marine_vending.dm [hunk 3]
```diff
@@ -757,16 +755,14 @@
 		/obj/item/reagent_containers/food/snacks/mre_pack/meal6 = -1,
 		/obj/item/storage/box/MRE = -1,
 		/obj/item/reagent_containers/food/drinks/flask = -1,
+		/obj/item/reagent_containers/food/drinks/flask/marine = 10,
+		/obj/item/reagent_containers/food/snacks/mre_pack/meal5 = 15
```
Comment (BraveMole) [inline code/game/objects/machinery/vending/marine_vending.dm:759]: infinite here

File: code/game/objects/machinery/fire_alarm.dm [hunk 4]
```diff
@@ -113,13 +113,7 @@ FIRE ALARM
...
 				playsound(loc, 'sound/items/crowbar.ogg', 25, 1)
 				spawn(20)
-					var/obj/item/circuitboard/firealarm/circuit
-					if(!electronics)
-						circuit = new/obj/item/circuitboard/firealarm(loc)
-					else
-						circuit = new electronics(loc)
-						if(electronics.is_general_board)
-							circuit.set_general()
+					new/obj/item/circuitboard/firealarm(loc)
```
Comment (BraveMole) [inline code/game/objects/machinery/fire_alarm.dm:116]:
```suggestion
					new /obj/item/circuitboard/firealarm(loc)
```

File: code/game/objects/machinery/doors/airlock.dm [hunk 5]
```diff
@@ -391,11 +391,6 @@
 			else if(length(req_one_access))
 				AE.conf_access = req_one_access
 				AE.one_access = TRUE
-		else
-			AE = electronics
-			if(electronics.is_general_board)
-				AE.set_general()
-			AE.forceMove(loc)
 			electronics = null
```
Comment (BraveMole) [inline code/game/objects/machinery/doors/airlock.dm:394]: you forgot this?

File: code/game/objects/items/stacks/sheets/sheet_types.dm [hunk 6]
```diff
@@ -11,58 +11,20 @@
 * Metal
 */
 GLOBAL_LIST_INIT(metal_recipes, list ( \
```
Comment (BraveMole) [inline code/game/objects/items/stacks/sheets/sheet_types.dm:13]: Imo, make everything uselessl in a big recipe list (chairs, bed, and keep everything), semi usefull thing in another recipe list (wall grider, apc frame, metal rod, grenade casing etc.) in another recipe and the things you need in emergency diretly in the metal recipe (cades, wires)

**Review by BraveMole — CHANGES_REQUESTED** [review]

File: code/game/objects/machinery/vending/vending_types.dm (hunk 1 above)
Comment (Arctrooper8) [reply code/game/objects/machinery/vending/vending_types.dm:472]: Is there a way to make it give out full belts? Right now it only gives out empty belts which seems somewhat annoying.

Comment (Arctrooper8) [reply code/game/objects/machinery/vending/vending_types.dm:472]: Also I thought the stance was Mesons are bad? If I'm uninformed I'll readd them.

File: code/game/objects/items/stacks/sheets/sheet_types.dm (hunk 6 above)
Comment (Arctrooper8) [reply code/game/objects/items/stacks/sheets/sheet_types.dm:13]: So don't remove any recipes?

**Comment by Arctrooper8** [comment]
Metal menu has been updated based on feedback!
New Metal Menu

File: code/game/objects/machinery/vending/vending_types.dm (hunk 1 above)
Comment (Arctrooper8) [reply code/game/objects/machinery/vending/vending_types.dm:472]: I moved mesons into engi vend and a full belt into the clothing vendor

**Comment by Derrick2Electricboogaloo** [comment]
I actually agree with Brave on the recipes, keep the stuff but maybe separate them? I personally did not find the 'old' (current) menu to be that cumbersome.

**Comment by Arctrooper8** [comment]
> I actually agree with Brave on the recipes, keep the stuff but maybe separate them? I personally did not find the 'old' (current) menu to be that cumbersome.

Yeah that should be what the second metal menu is. I think it's missing metal bat, machine frame, and computer frame. But those are recipes that genuinely are never used.

File: code/game/objects/machinery/vending/marine_vending.dm [hunk 7]
```diff
@@ -909,7 +900,7 @@
 	var/obj/item/reagent_containers/blood/temp_path
 	var/datum/vending_product/R
 	var/blood_type
-	for(R in (product_records + hidden_records + coin_records))
+	for(R in (product_records + coin_records))
```
Comment (BraveMole) [inline code/game/objects/machinery/vending/marine_vending.dm:903]:
```suggestion
	for(var/datum/vending_product/R AS in (product_records + coin_records))
```

File: code/game/objects/machinery/vending/marine_vending.dm [hunk 8]
```diff
@@ -909,7 +900,7 @@
 	var/obj/item/reagent_containers/blood/temp_path
 	var/datum/vending_product/R
```
Comment (BraveMole) [inline code/game/objects/machinery/vending/marine_vending.dm:901]:
```suggestion
```

**Review by BraveMole — CHANGES_REQUESTED** [review]

**Review by BraveMole — APPROVED** [review]

**Pull request merged**